domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientIT.java

```diff
@@ -2224,51 +2221,206 @@ public void getDistinctOn(TestContext context) throws IOException {
+      });
+  }
+
+  @Test
+  public void getCQLWrapperJsonbField(TestContext context) throws IOException, FieldException {
+    final String tableDefiniton = "id UUID PRIMARY KEY , jsonb JSONB NOT NULL, distinct_test_field TEXT";
+    createTableWithPoLines(context, MOCK_POLINES_TABLE, tableDefiniton);
+
+    CQL2PgJSON cql2pgJson = new CQL2PgJSON(MOCK_POLINES_TABLE + ".jsonb");
+    List<FacetField> facets = null;
+
+    CQLWrapper cqlWrapper = new CQLWrapper(cql2pgJson, "cql.allRecords=1");
+    Async async = context.async();
+    postgresClient.get(MOCK_POLINES_TABLE, Object.class, "jsonb",
+      cqlWrapper, true, true, false, null, facets, handler -> {
+        context.assertTrue(handler.succeeded());
+        ResultInfo resultInfo = handler.result().getResultInfo();
+        context.assertEquals(6, resultInfo.getTotalRecords());
+        async.complete();
+      });
+    async.awaitSuccess();
+
+    String distinctOn = "jsonb->>'order_format'";
+    cqlWrapper = new CQLWrapper(cql2pgJson, "cql.allRecords=1");
+    postgresClient.get(MOCK_POLINES_TABLE, Object.class, "jsonb",
+      cqlWrapper, true, true, false, distinctOn, facets, handler -> {
+        context.assertTrue(handler.succeeded());
+        ResultInfo resultInfo = handler.result().getResultInfo();
+        context.assertEquals(4, resultInfo.getTotalRecords());
+        async.complete();
+      });
+    async.awaitSuccess();
+  }
+
+  @Test
+  public void getCQLWrapperNoFacets(TestContext context) throws IOException, FieldException {
+    final String tableDefiniton = "id UUID PRIMARY KEY , jsonb JSONB NOT NULL, distinct_test_field TEXT";
+    createTableWithPoLines(context, MOCK_POLINES_TABLE, tableDefiniton);
+
+    CQL2PgJSON cql2pgJson = new CQL2PgJSON(MOCK_POLINES_TABLE + ".jsonb");
+    List<FacetField> facets = null;
+    CQLWrapper cqlWrapper = new CQLWrapper(cql2pgJson, "cql.allRecords=1");
+    Async async = context.async();
+    postgresClient.get(MOCK_POLINES_TABLE, Object.class, "*",
+      cqlWrapper, true, true, false, null, facets, handler -> {
+        context.assertTrue(handler.succeeded());
+        ResultInfo resultInfo = handler.result().getResultInfo();
+        context.assertEquals(6, resultInfo.getTotalRecords());
+        List<Facet> retFacets = resultInfo.getFacets();
+        context.assertEquals(0, retFacets.size());
+        async.complete();
+      });
+    async.awaitSuccess();
+
+    String distinctOn = "jsonb->>'order_format'";
+    cqlWrapper = new CQLWrapper(cql2pgJson, "cql.allRecords=1");
+    postgresClient.get(MOCK_POLINES_TABLE, Object.class, "*",
+      cqlWrapper, true, true, false, distinctOn, facets, handler -> {
+        context.assertTrue(handler.succeeded());
+        ResultInfo resultInfo = handler.result().getResultInfo();
+        context.assertEquals(4, resultInfo.getTotalRecords());
+        List<Facet> retFacets = resultInfo.getFacets();
+        context.assertEquals(0, retFacets.size());
+        async.complete();
```

Comment:
You cannot reuse a succeded Async.

Reply:
There is no need for Async, you can run these three get requests in parallel and make use of context.asyncAssertSuccess.